dev/docs.go

```diff
@@ -52,11 +53,25 @@ func (d Docs) Server() *dagger.Container {
 }
 
 // Lint documentation files
-func (d Docs) Lint(ctx context.Context) error {
+func (d Docs) Lint(ctx context.Context) (rerr error) {
+	ctx, span := Tracer().Start(ctx, "lint Dagger documentation")
+	defer func() {
+		if rerr != nil {
+			span.SetStatus(codes.Error, rerr.Error())
+		}
+		span.End()
+	}()
```

Comment:
I understand if you want to skip it to stay familiar with the raw OTel SDK, but there's a helper available for this provided by the Dagger Go SDK:

```go
import (
	"github.com/dagger/dagger/dev/internal/telemetry"
)
```

```go
defer telemetry.End(span, func() error { return rerr })
```

We're not skipping `Tracer()` so this would be consistent with using the helpers, but yeah no big deal if you prefer not to lean on it.

Reply:
@vito any chance the helper could receive the error value, so I can just pass it through? Would save me 1 line in each diff.

Or even better, just pass it through for me: `defer telemetry.End(span)`

Reply:
@shykes The trick is the helper needs the `error` value at the time the function returns, so we need to capture a reference to it so the `defer` doesn't just call it with `nil` (`rerr` value at the time).

Right now it's a closure, but I regret that now - I think a pointer would also work:

```go
defer telemetry.End(span, &rerr)
```

I don't think there's any way to magically grab the error value in Go.